### PR TITLE
feature: in string validation

### LINKED ISSUE #18: Feat: String validations
With this task must be allowed to use the following validations in string fields:

- eq (equal): must be equal to the specified value
- eq_ignore_case (equal ignoring case): must be equal to the specified value (ignoring case)
- ne (not equal): must not be equal to the specified value
- ne_ignore_case (not equal ignoring case): must not be equal to the specified value (ignoring case)
- len (length): must have the following length
- max (max): must have no more than max characters
- min (min): must have no less than min characters
- in (in): must be on of the following values
- nin (not in): must not be on of the following values
- required (required): is required

### PR DESCRIPTION
**Pull request by alexgarzao**
Related to #18 

This PR introduces enhancements to validation handling, including support for multi-value validations such as "in" and improvements to error formatting and testing. The most significant changes involve extending validation logic, refactoring related test structures, and adding comprehensive unit tests to ensure robustness.

### Enhancements to validation logic:
- **Support for "in" validation**: Added the `FieldIn` field to `StringType` with validation for specific values (`ab`, `bc`, `cd`). Updated `StringTypeValidate` to enforce this validation. [[1]](diffhunk://#diff-4f0802d7c2e573b0e4aaa631328677086e14b794beeb22c7fa1219a7bb92c4cbR13) [[2]](diffhunk://#diff-1d142222f7677210e08405f7a33515144de5511e39890fe4a562583328b24724R44-R47)
- **Refactored validation error creation**: Modified `NewValidationError` to support formatted error messages using variadic arguments.

### Improvements to test structures:
- **Refactored `TestElements`**: Replaced `rightOperand` with `rightOperands` to support multiple values in validation tests. Updated associated test cases for numeric and string fields. [[1]](diffhunk://#diff-13fb4a2c2bb5fa8bda222776db3ec7f7afe50f4cf7f374db36a2031898e5462cL30-R30) [[2]](diffhunk://#diff-13fb4a2c2bb5fa8bda222776db3ec7f7afe50f4cf7f374db36a2031898e5462cL45-R45) [[3]](diffhunk://#diff-3b9d1a12858980987436bb3f555f8f3861f65a83e810b28aaffda521dc11c071L27-R27) [[4]](diffhunk://#diff-3b9d1a12858980987436bb3f555f8f3861f65a83e810b28aaffda521dc11c071L118-R147)
- **Validation condition generation**: Enhanced `IfCode` to handle multiple operands for validations like "in".

### Comprehensive unit tests:
- **New tests for "in" validation**: Added unit tests for parsing and validating multi-value tags with various formats (e.g., space-separated, quoted values). [[1]](diffhunk://#diff-8547d0bda459f189f0eb9cc5e0219b7fbe38b4dbc7fb656e4c453f20a177a281R1-R168) [[2]](diffhunk://#diff-624caa67bdb0ba682f775f0fa90bbfa9ff8df232412316dd08381274997123b6R1-R152)
- **Extended test coverage for `GetTestElements`**: Added test cases for handling multi-value validations and edge cases. [[1]](diffhunk://#diff-cf43f1f1e3cf48a3ecbe6cc814e3a5e2934274f2bdf4e0e0d9386c093e34f91fR1-R55) [[2]](diffhunk://#diff-871326416a200a07b2b5ef890daed108971c94f739c84d93d27975fda2643ee4R41-R53)

These changes collectively improve the flexibility and reliability of the validation framework while ensuring robust test coverage.

### CODE REVIEW

**Review by felipeagger — APPROVED** [review]

File: validgen/parser_validation.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+		"required":        ZERO_VALUE,
+		"gte":             ONE_VALUE,
+		"lte":             ONE_VALUE,
+		"min":             ONE_VALUE,
+		"max":             ONE_VALUE,
+		"eq_ignore_case":  ONE_VALUE,
+		"len":             ONE_VALUE,
+		"neq":             ONE_VALUE,
+		"neq_ignore_case": ONE_VALUE,
+		"oneof":           MANY_VALUES,
```
[the remark this answers is not in the document]
Comment (fabriciogpp) [reply validgen/parser_validation.go:36]: Inclusive abre a possibiliade de um `nin` (not in).
